enterprise/server/remote_execution/containers/podman/podman_test.go

```diff
@@ -0,0 +1,103 @@
+package podman_test
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/podman"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/platform"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/oci"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	_ "github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/containers/docker"
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+type SlowCommandRunner struct {
+	commandsRun *atomic.Int32
```

Comment:
nit: this can be an `atomic.Int32` (non-pointer) so you don't have to construct it manually. The test can directly check `runner.commandsRun`

Reply:
That leads to a copying lock error on line 40 where the command runner is copied in the environment. I could make that a pointer, but I'd rather leave the pointer messiness in the test if that's alright with you.

Reply:
You can avoid that error by changing the function signature on line 27 to be a pointer receiver

```
func (r *SlowCommandRunner) ...
```

then on line 40, declare SlowCommandRunner using a pointer:

```
commandRunner := &SlowCommandRunner{}
```

To me this is not just a workaround, but actually the most natural way to do it - for "business logic" structs which have methods and implement interfaces, I've found it's almost always desired to pass by reference and not by value. Otherwise, if you want pass-by-reference semantics in your method calls, then you have to declare all fields as pointers, which is a lot more tedious/complex than just declaring the struct itself as a pointer.

The only times I've found it makes sense to pass structs by value is for very small "tuple"-like structures consisting only of primitives, such as username/password combos, or options structs with a few int/bool config options.

Reply:
Ahh, I didn't realize you could mix-and-match pointer versus non-pointer receivers for different implementations of an interface and was a bit hesitant to make the real instance receive a pointer (though I guess that doesn't really matter either), but what you suggested works so I changed it to that.